### ISSUE TITLE
[3.3.5][6.x] Problem with spell target indicator and npcs/pets on elevator.

### ISSUE THREAD

**Issue by burbur** [TrinityCore/TrinityCore#14963]
Hi there. 

Rev: 3d6abdb2ff3b4f672890ce9762fa5eefbcf111c8
Db: TDB 6.01 

I've noticed a general issue with npcs (pets for now) on the elevators. 
When moving on an elevator (Orgrimmar or Undercity ones for example) the pet doesn't "follow you" but remains on the last position it had. 
For example if we take the undercity elevator, with two floors, when you go on the elevator on the first floor, as the elevator starts to climb the pet will stay where it is, under the elevator.
Instead, when you move on the elevator on the upper floor, as it starts to go down the pet just stands in the air, without moving, as if his Z position will remain unchanged. 
TL; DR;  pets cannot use elevators like players. I guess it's related to every npc. 

Other issue is related to spell target indicator (like Death & decay, Healing Rain...), the one that allow you to choose the destination of the spell. 
When you are on the elevator (or any generic transport object like trap doors) you won't see the indicator when using one of those spell. I've just tried with some spells. If you try to cast them you get a generic not in LoS. 

I've done a video on retail (6.2.20182) showing this (undercity's elevator) with my spider pet and "Move To" command that has the spell indicator i've mentioned early:
https://vid.me/KJNg

**Comment by Keader** [comment]
The problem with using spells in elevators is also in 3.3.5a. Apparently coordinates the elevators are recognized as 0,0,0

**Comment by Nyeriah** [comment]
Yeah elevators are weird on both branches. Additionally, if you jump on an elevator the height is supposed to be calculated from the elevator's current position, but currently if you jump while on them you'll be dropped off and fall to death.

**Comment by burbur** [comment]
@Nyeriah didn't find the issue about falling when jumping on 6.x

Anyway this is an important thing from 4.x because some bosses are played on platforms: Nefarian End's encounter is on an elevator, Cho'gall in Bastion of twilight has a trap door in the middle of the room and Blackhand, last boss, in Blackrock Foundry (Draenor) has two platforms that get destroyed as he has 66/33% hp... something like Trial of the Crusader where you go down after Twin Val'kyrs. In this last case maybe it will work because objects should be the same as the ToC one.

**Comment by Keader** [comment]
Ring of Valor(arena) too :v:

**Comment by akrom23** [comment]
Elevators, ships in IoC, Deeprun Tram (GAMEOBJECT_TYPE_TRANSPORT) don't update unit's position.

**Comment by Aokromes** [comment]
This bug is still valid?

**Comment by Aokromes** [comment]
No answer for 5 months, closing, if this bug is still valid open a new ticket.

**Comment by ariel-** [comment]
Known issue, transports do not update positions serverside.... something like https://github.com/Kittnz/Sunwell/blob/master/src/server/game/Entities/Transport/Transport.h#L111 should be implemented (support for GAMEOBJECT_TYPE_TRANSPORT position update. 

Only problem with this is SWC relies on keeping grids loaded for transports. See #17360 for further discussion on that issue.
